src/test/Mission.test.js

```diff
@@ -0,0 +1,51 @@
+import missionReducer, {
+  initialState,
+  joinMissionReducer,
+  fetchMission,
+} from '../redux/missionSlice/Mission';
+
+describe('missionSlice', () => {
+  it('should return the initial state', () => {
+    const nextState = missionReducer(undefined, {});
+    expect(nextState).toEqual(initialState);
+  });
+
+  it('should handle joinMissionReducer', () => {
+    const initialStateWithData = {
+      data: [
+        { mission_id: '1', reserved: false },
+        { mission_id: '2', reserved: false },
+      ],
+    };
+    const nextState = missionReducer(
+      initialStateWithData,
+      joinMissionReducer({ missionIdToJoin: '1', reserved: false }),
+    );
+    expect(nextState.data[0].reserved).toBe(true);
+    expect(nextState.data[1].reserved).toBe(false);
+  });
+
+  it('should handle fetchMission.pending', () => {
+    const nextState = missionReducer(undefined, fetchMission.pending);
+    expect(nextState.isLoading).toBe(true);
+  });
+
+  it('should handle fetchMission.fulfilled', () => {
+    const missionData = [
+      { mission_id: '1', mission_name: 'Mission 1' },
+      { mission_id: '2', mission_name: 'Mission 2' },
+    ];
+    const nextState = missionReducer(
+      undefined,
+      fetchMission.fulfilled(missionData),
+    );
+    expect(nextState.isLoading).toBe(false);
+    expect(nextState.data).toEqual(missionData);
+  });
+
+  it('should handle fetchMission.rejected', () => {
+    const nextState = missionReducer(undefined, fetchMission.rejected);
+    expect(nextState.isLoading).toBe(false);
+    expect(nextState.isError).toBe(true);
+  });
```

Comment:
- [ ] Kindly write tests for your components using React Testing Library. Testing your components ensures that they function properly. You can read these articles to learn how to use React testing library 😎 💻:

https://testing-library.com/docs/react-testing-library/intro/
https://www.robinwieruch.de/react-testing-library/